packages/gatsby-source-nacelle/src/utils/createRemoteImageFileNode.js

```diff
@@ -0,0 +1,120 @@
+const get = require('lodash.get');
+const set = require('lodash.set');
+
+function getNodeMedia(node, nodeMedia) {
+  // given an object and a property (or array of properties to
+  // access a nested property), return the value of that property
+  if (Array.isArray(nodeMedia)) {
+    return get(node, nodeMedia, null);
+  }
+
+  return node[nodeMedia];
+}
+
+function setNodeMedia(node, nodeMedia, fileNode, newField = 'remoteImage') {
+  // given an object, a property (or array of properties to access
+  // a nested property), and a file node, associate the '@link File'
+  // field with the file node
+  if (Array.isArray(nodeMedia)) {
+    set(node, [...nodeMedia, newField], fileNode.id);
+  } else {
+    node[nodeMedia][newField] = fileNode.id;
+  }
+
+  return node;
+}
+
+async function createFileNode(
+  node,
+  nodeMedia,
+  gatsbyApi,
+  { isImage = () => true, imageProperties = ['src', 'url'] }
+) {
+  let createRemoteFileNode;
+  try {
+    createRemoteFileNode = require('gatsby-source-filesystem')
+      .createRemoteFileNode;
+  } catch (err) {
+    createRemoteFileNode = null;
+  }
+
+  if (createRemoteFileNode) {
```

Comment:
Similar to previous comment -- should be do this inside the above `try` block rather than putting it inside another conditional. Additionally, I wonder if there's an opportunity to create some smaller functions here - this code can get hard to follow when you have a `try => if => try => if => if` in a single function, and would likely be easy to misunderstand / mess up in the future.